src/js/utils/date-utils.js

```diff
@@ -0,0 +1,258 @@
+
+        /**
+         *
+         * Returns a new time that is offset by the requested timezone.
+         *
+         * @param {Object} dateTimeObject A moment.js datetime object.
+         * @param {string} timezone A timezone representation, e.g. 'America/New_York'.
+         * @return {object} A moment.js datetime object in a determined timezone.
+         */
+        localizeTime = function(dateTimeObject, timezone) {
+            var localTime;
+            if (isValid(dateTimeObject) && isValid(timezone)) {
+                localTime = dateTimeObject.tz(timezone);
+            } else if (isValid(dateTimeObject)) {
+                localTime = dateTimeObject.tz(getSysTimezone());
+            } else {
+                localTime = '';
+            }
+            return localTime;
+        };
+
+        /**
+         *
+         * Helper function to determine system timezone, and fallback to UTC
+         * Determines timezone from argument, falling back to system/browser settings, then UTC.
+         *
+         * @return {string} Will attempt to return a determined timezone string, with a fallback to UTC.
+         *
+         */
+        getSysTimezone = function() {
+            var timezone = momentTZ.tz.guess();
+            if (!isValid(timezone)) {
+                timezone = DEFAULT_TIMEZONE;
+            }
+            return timezone;
+        };
+
+        /**
+         *
+         * Generate a timezone-aware and language-aware string.
+         *
+         * This includes a param for standard formatting constants.
+         *
+         * Browser preferred language is currently overridden by platform-set language preference.
+         * If no preference is set, the function attempts to determine the locale settings
+         * (e.g. en-US) from the browser preferences.
+         *
+         * This function will default to UTC/en-US display in default format in the case of
+         * an unresolvable language or timezone
+         *
+         * preferred_language = 'en' yields Oct. 14, 2016 09:00 BST
+         *
+         * preferred_language = 'ru' yields 14 окт 2016 г. 09:00 BST
+         *
+         * @param {object} localTime A moment.js datetime object offset to a determined timezone.
+         * @param {string} language an ISO 639-1 language code (e.g. 'en-US', 'fr')
+         * @param {string} format (optional) A desired date format.
+         * @return {string} A language/timezone aware localized string
+        */
+        getDisplayTime = function(localTime, language, format) {
+            var displayTime = '';
+            var displayLang = getDisplayLang(language);
+            var displayFormat = getDisplayFormat(format);
+
+            if (isValid(displayLang) && isValid(displayFormat)) {
+                if (moment(localTime).isValid()) {
+                    displayTime = localTime.locale(displayLang).format(displayFormat);
+                } else {
+                    displayTime = '';
+                }
```

Comment:
Is this else clause needed?  `displayTime` is already defaulted to the empty string above.  So can remove these 3 lines, right?